### PR TITLE
Updated script to deploy simulation using PowerShell Az and updated I…

### PR DESCRIPTION
**Pull request by markusstuhler**
…oT Edge Deployment

### CODE REVIEW

File: tools/e2etesting/DeploySimulation.ps1 [hunk 1]
```diff
@@ -1,116 +1,160 @@
...
-az config set extension.use_dynamic_install=yes_without_prompt
-
-# Powershell-Az module (which is automatically authorized by Powershell) 
-# don't support IoT Hub Device Identity methods, we ned to use az cli
-if(-Not ([string]::IsNullOrEmpty($client_id) -or [string]::IsNullOrEmpty($client_secret) -or [string]::IsNullOrEmpty($tenantId))) {
-    az login --service-principal --username  $client_id --password $client_secret --tenant $tenantId
-} else {
-    Write-Warning "username or password or tenantId for az login are not provided - skipped"
+if (!$ResourceGroupName) {
+    Write-Error "ResourceGroupName not set."
```
Comment (koepalex) [inline tools/e2etesting/DeploySimulation.ps1:11]: add "-ErrorAction Stop" ?

File: tools/e2etesting/DeploySimulation.ps1 [hunk 2]
```diff
@@ -1,116 +1,160 @@
...
+if (!$context) {
+    Write-Host "Logging in..."
+    Login-AzAccount -Tenant $TenantId
+    $context = Get-AzContext
+}
+
+$resourceGroup = Get-AzResourceGroup -Name $resourceGroupName
+
+if (!$resourceGroup) {
+    Write-Error "Could not find Resource Group '$($ResourceGroupName)'."
```
Comment (koepalex) [inline tools/e2etesting/DeploySimulation.ps1:30]: add "-ErrorAction Stop" instead of add return?

File: tools/e2etesting/DeploySimulation.ps1 [hunk 3]
```diff
@@ -1,116 +1,160 @@
...
-
-# Store Device Id within KeyVault to be useable by rerun without deployment (USeExisting=true)
-$secretvalue = ConvertTo-SecureString $edgeIdentity.deviceId -AsPlainText -Force
-$secret = Set-AzKeyVaultSecret -VaultName $keyVaultName -Name 'iot-edge-device-id' -SecretValue $secretvalue
-if (-Not $secret.Enabled) {
-    Write-Error "Couldn't store device identity in KeyVault" -ErrorAction Stop
+$iotHub = Get-AzIotHub -ResourceGroupName $ResourceGroupName
+
+if ($iotHub.Count -ne 1) {
+    Write-Error "IotHub could not be automatically selected in Resource Group '$($ResourceGroupName)'."
```
Comment (koepalex) [inline tools/e2etesting/DeploySimulation.ps1:51]: add "-ErrorAction Stop" instead of add return?

File: tools/e2etesting/DeploySimulation.ps1 [hunk 4]
```diff
@@ -1,116 +1,160 @@
...
-$deviceTwin | Add-Member -MemberType NoteProperty -Value $tags -Name "tags"
-$configFileName = New-TemporaryFile
-($deviceTwin | ConvertTo-Json -Depth 10) | Out-File -FilePath $configFileName -Encoding UTF8
-$out = az iot hub device-twin replace --device-id $edgeIdentity.deviceId --hub-name $iotHub.Name --json $configFileName
-Write-Host "Added Industrial IoT tags to Device Twin of edge device"
-
-$vmpassword = -join ((65..90) + (97..122) + (33..38) + (40..47) + (48..57)| Get-Random -Count 20 | % {[char]$_})
-$numberOfSimulations = 18
+Write-Host "Adding/Updating KeyVault-Secret 'iot-edge-device-id' with value '$($edgeIdentity.Id)'..."
+Set-AzKeyVaultSecret -VaultName $keyVault.VaultName -Name 'iot-edge-device-id' -SecretValue (ConvertTo-SecureString $edgeIdentity.Id -AsPlainText -Force) | Out-Null
```
Comment (koepalex) [inline tools/e2etesting/DeploySimulation.ps1:80]: Why not check, if this was successfully?

File: tools/e2etesting/DeploySimulation.ps1 (hunk 1 above)
Comment (markusstuhler) [reply tools/e2etesting/DeploySimulation.ps1:13]: I added a global setting for this.

File: tools/e2etesting/DeploySimulation.ps1 [hunk 5]
```diff
@@ -1,116 +1,160 @@
...
+
+Write-Host "Updating 'os' and '__type__'-Tags in Device Twin..."
+Update-AzIotHubDeviceTwin -ResourceGroupName $ResourceGroupName -IotHubName $iotHub.Name -DeviceId $edgeIdentity.Id -Tag @{ "os" = "Linux"; "__type__" = "iiotedge"} | Out-Null
```
Comment (koepalex) [inline tools/e2etesting/DeploySimulation.ps1:83]: Why not check, if this was successfully?

File: tools/e2etesting/DeploySimulation.ps1 (hunk 4 above)
Comment (markusstuhler) [reply tools/e2etesting/DeploySimulation.ps1:80]: -> global Stop handler.

File: tools/e2etesting/DeploySimulation.ps1 [hunk 6]
```diff
@@ -1,116 +1,160 @@
...
+$dnsPrefix = "e2etesting-edgevm-" + $testSuffix
+
+Write-Host "Using DNS prefix: $($dnsPrefix)"
+
+$edgeParameters = @{
+    "dnsLabelPrefix" = $dnsPrefix
+    "adminUsername" = $edgeVmUsername
+    "deviceConnectionString" = $edgeDeviceConnectionString
+    "authenticationType" = "password"
+    "adminPasswordOrKey" = $edgeVmPassword
```
Comment (koepalex) [inline tools/e2etesting/DeploySimulation.ps1:131]: setting vmSize to "Standard_D2s_v3" is missing

File: tools/e2etesting/DeploySimulation.ps1 [hunk 7]
```diff
@@ -1,116 +1,157 @@
...
-    $tenantId
+    $ResourceGroupName,
+    [int]
+    $NumberOfSimulations = 18,
+    [Guid]
+    $TenantId
 )
 
-az config set extension.use_dynamic_install=yes_without_prompt
+$ErrorActionPreference = "Stop"
```
Comment (koepalex) [inline tools/e2etesting/DeploySimulation.ps1:10]: Add comment, what it does

File: tools/e2etesting/DeploySimulation.ps1 (hunk 5 above)
Comment (markusstuhler) [reply tools/e2etesting/DeploySimulation.ps1:83]: If not, it throws an error and the script stops.

File: tools/e2etesting/DeploySimulation.ps1 (hunk 6 above)
Comment (markusstuhler) [reply tools/e2etesting/DeploySimulation.ps1:146]: Reintroduced parameter.

File: tools/e2etesting/DeploySimulation.ps1 (hunk 7 above)
Comment (markusstuhler) [reply tools/e2etesting/DeploySimulation.ps1:15]: Done.

**Pull request merged**